samples/sql-persistence/transitioning-correlation-ids/SqlPersistence_2/Phase1/EndpointOracle/Program.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Persistence.Sql;
+using Oracle.ManagedDataAccess.Client;
+
+partial class Program
+{
+    static void Main()
+    {
+        AsyncMain().GetAwaiter().GetResult();
+    }
+
+    static async Task AsyncMain()
+    {
+        Console.Title = "EndpointOracle";
+
+        var endpointConfiguration = new EndpointConfiguration("EndpointOracle");
+
+        endpointConfiguration.SendFailedMessagesTo("error");
+        endpointConfiguration.UseSerialization<JsonSerializer>();
+        endpointConfiguration.EnableInstallers();
+        var persistence = endpointConfiguration.UsePersistence<SqlPersistence>();
+        var password = Environment.GetEnvironmentVariable("OraclePassword");
+        if (string.IsNullOrWhiteSpace(password))
+        {
+            throw new Exception("Could not extract 'OraclePassword' from Environment variables.");
+        }
+        var username = Environment.GetEnvironmentVariable("OracleUserName");
+        if (string.IsNullOrWhiteSpace(username))
+        {
+            throw new Exception("Could not extract 'OracleUserName' from Environment variables.");
+        }
+        var connection = $"Data Source=localhost;User Id={username}; Password={password}; Enlist=false";
```

Comment:
Oracle supports DTC, but Enlist=true (or omitting entirely) seems to enlist prematurely. When we do that, in acceptance tests I get subscription messages going to the error queue with:

> Failed to ExecuteNonQuery
> Oracle.ManagedDataAccess.Client.OracleException: ORA-02089: COMMIT is not allowed in a subordinate session

Now, the subscribe code does have a commit/rollback thing in it:

```
begin
    insert into Publisher1615587009_SS
    (
        MessageType,
        Subscriber,
        Endpoint,
        PersistenceVersion
    )
    values
    (
        :MessageType,
        :Subscriber,
        :Endpoint,
        :PersistenceVersion
    );
    commit;
exception
    when DUP_VAL_ON_INDEX
    then ROLLBACK;
end;
```

I suppose we could try rewriting that, but with Enlist=false, we still have distributed transactions at appropriate times and everything works.